server/src-lib/Hasura/RQL/Types/RemoteSchema.hs

```diff
@@ -111,3 +116,84 @@ validateRemoteSchemaDef env (RemoteSchemaDef mUrl mUrlEnv hdrC fwdHdrs mTimeout)
     hdrs = fromMaybe [] hdrC
 
     timeout = fromMaybe 60 mTimeout
+
+data RemoteSchemaPermissionDefinition
+  = RemoteSchemaPermissionDefinition
+  { _rspdSchema    :: !G.SchemaDocument
+  }  deriving (Show, Eq, Lift, Generic)
+instance NFData RemoteSchemaPermissionDefinition
+instance Cacheable RemoteSchemaPermissionDefinition
+instance Hashable RemoteSchemaPermissionDefinition
+
+instance J.FromJSON RemoteSchemaPermissionDefinition where
+  parseJSON = J.withObject "RemoteSchemaPermissionDefinition" $ \obj -> do
+    schema <- obj J..: "schema"
+    flip (J.withText "schema") schema $ \t ->
+      let schemaDoc = J.fromJSON $ J.String t
+      in
+      case schemaDoc of
+        J.Error err -> fail err
+        J.Success a -> return $ RemoteSchemaPermissionDefinition a
```

Comment:
I.... think you can discard this entirely? You're trying to parse something that has a `FromJSON` instance. Can't you just do the following?
```
\obj -> fmap RemoteSchemaPermissionDefinition $ obj J..: "schema"
```

Reply:
Oh god, what was I thinking 😓 . Thanks for the suggestion!

Reply:
I've implemented the suggestion